### ISSUE TITLE
azure-arm invalid parameter dataDisk.name

### ISSUE THREAD

**Issue by stephanwehr** [hashicorp/packer#9249]
### Overview of the Issue

After updating from version 1.5.5 to 1.5.6 it is no longer possible to create a specific VM image on Azure. The build is still possible with version 1.5.5.

Error message:
```
==> azure-arm: Deploying deployment template ...
==> azure-arm:  -> ResourceGroupName : 'pkr-Resource-Group-to02kx6mfx'
==> azure-arm:  -> DeploymentName    : 'pkrdpto02kx6mfx'
==> azure-arm: ERROR: -> DeploymentFailed : At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/DeployOperations for usage details.
==> azure-arm: ERROR:   -> BadRequest
==> azure-arm: ERROR:   -> InvalidParameter : The entity name 'dataDisk.name' is invalid according to its validation rule: ^[^_\W][\w-._]{0,79}(?<![-.])$.
```

### Reproduction Steps

Start a build using the Buildfile below.
Provide the variables as parameters.

### Packer version

Version 1.5.6

### Operating system and Environment details

Ubuntu Linux 18.04

### Simplified Packer Buildfile

```
{
  "variables": {
    "client_id": "00000000-0000-0000-0000-000000000000",
    "client_secret": "00000000-0000-0000-0000-000000000000",
    "tenant_id": "00000000-0000-0000-0000-000000000000",
    "subscription_id": "00000000-0000-0000-0000-000000000000",
    "resource_group_name": "myname-rg",
    "image_name": "myname-img",
    "location": "mylocation",
    "image_sku": "18.04-LTS",
    "vm_size": "Standard_B4ms",
    "info_tag": "myinfo"
  },
  "builders": [{
    "type": "azure-arm",

    "shared_image_gallery_timeout": "180m",
    "polling_duration_timeout": "180m",

    "subscription_id": "{{user `subscription_id`}}",
    "client_id": "{{user `client_id`}}",
    "client_secret": "{{user `client_secret`}}",
    "tenant_id": "{{user `tenant_id`}}",

    "managed_image_resource_group_name": "{{user `resource_group_name`}}",
    "managed_image_name": "{{user `image_name`}}",

    "os_type": "Linux",
    "os_disk_size_gb": 31,
    "image_publisher": "Canonical",
    "image_offer": "UbuntuServer",
    "image_sku": "{{user `image_sku`}}",

    "disk_caching_type": "ReadWrite",
    "disk_additional_size": [31],

    "location": "{{user `location`}}",
    "vm_size": "{{user `vm_size`}}",

    "azure_tags": {
        "CreatedBy": "Packer",
        "Info": "{{user `info_tag`}}"
    }
  }],
  "provisioners": [
    ...
  ]
}
```

**Comment by nywilken** [comment]
Thanks for reaching out we will take a look at this in time for the next release.

**Comment by nywilken** [comment]
Hello again :wave: I went ahead and opened a PR with a possible fix for this issue #9251. There are some test binaries available [here](https://app.circleci.com/pipelines/github/hashicorp/packer/5077/workflows/06ebefe3-7031-4fe3-b734-924d8e662e35/jobs/52207/artifacts) if you would like to give the fix a try.
